flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/StreamingStartingStrategy.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.source;
+
+/**
+ * Starting strategy for streaming execution. Note that starting snapshot is exclusive in the incremental
+ * split discovery mode. Files appended by the starting snapshot aren't included in split discovery.
+ */
+public enum StreamingStartingStrategy {
```

Comment:
Is this just a strategy for unbounded reading ? And will there be a strategy for bounded reading, e.g.  `BatchStartingStragegy` ?

Reply:
It is just a streaming strategy. Streaming source can also be bounded, which is also the case for Flink kafka source. As mentioned in another comment, it is reasonable to have the Iceberg source stop at a future timestamp. It is not implemented in this PR. We can discuss the feature in the future if people want it useful.

For batch scan, we won't need any starting strategy. We would just need to specify the filters.